File: LeetCode June Challenge/ShortEncodingofWords.js
var minimumLengthEncoding = function(words) {
    let mySet = new Set(words);
    for(let Word of words){
        size = Word.length;
        for(let i = 1; i < size; ++i){
            let str = Word.slice(i, size);
            if(mySet.has(str)){
                mySet.delete(str);
            }
        }
    }
    let ans = 0;
    for(let newSet of mySet){
        ans += newSet.length + 1;
    }
    return ans;
};
let words = ["k"];
console.log(minimumLengthEncoding(words));

File: LeetCode June Challenge/ShortEncodingofWords.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int minimumLengthEncoding(vector<string>& words) {
        unordered_set<string> mySet(words.begin(), words.end());
        for(auto ch : words){
            int size = ch.size();
            for(int i = 1; i < size; ++i){
                string str = ch.substr(i, size);
                if(mySet.find(str) != mySet.end()){
                    mySet.erase(str);
                }
            }
        }
        int ans = 0;
        for(auto myWords : mySet){
            ans += myWords.size() + 1;
        }
        return ans;
    }
};
int main(){
    Solution s;
    vector<string> words = {""};
    cout<<s.minimumLengthEncoding(words)<<endl;
    return 0;

}